pkg/jobrunaggregator/jobrunaggregatorlib/big_query.go

```diff
@@ -81,38 +81,35 @@ func NewDryRunInserter(out io.Writer, table string) BigQueryInserter {
 func (d dryRunInserter) Put(ctx context.Context, src interface{}) (err error) {
 	srcVal := reflect.ValueOf(src)
 	if srcVal.Kind() != reflect.Slice {
-		fmt.Fprintf(d.out, "INSERT into %v: %v\n", d.table, src)
+		logrus.Debugf("INSERT into %s: %v", d.table, src)
 		return
 	}
 
 	if srcVal.Len() == 0 {
 		return
 	}
 
-	buf := &bytes.Buffer{}
-	fmt.Fprintf(buf, "BULK INSERT into %v\n", d.table)
+	logrus.Debugf("BULK INSERT into %s", d.table)
 	for i := 0; i < srcVal.Len(); i++ {
 
 		switch s := srcVal.Index(i).Interface().(type) {
 		case *jobrunaggregatorapi.TestRunRow:
-			fmt.Fprintf(buf, "\tINSERT into %v: %#v\n", d.table, s)
+			logrus.Debugf("   INSERT into %s: %#v", d.table, s)
 
 		case *jobrunaggregatorapi.JobRunRow:
-			fmt.Fprintf(buf, "\tINSERT into %v: name=%v, jobname=%v, status=%v\n", d.table, s.Name, s.JobName, s.Status)
+			logrus.Debugf("   INSERT into %s: name=%s, jobname=%s, status=%s", d.table, s.Name, s.JobName, s.Status)
 
 		case *jobrunaggregatorapi.BackendDisruptionRow:
-			fmt.Fprintf(buf, "\tINSERT into %v: %#v\n", d.table, s)
+			logrus.Debugf("   INSERT into %s: %#v", d.table, s)
 
 		case jobrunaggregatorapi.JobRow:
-			fmt.Fprintf(buf, "\tINSERT into %v: JobName=%v\n", d.table, s.JobName)
+			logrus.Debugf("   INSERT into %s: JobName=%s\n", d.table, s.JobName)
 
 		default:
-
 			// If we don't know the type, output something generic.
-			fmt.Fprintf(buf, "\tINSERT into %v: %#v\n", d.table, s)
+			logrus.Debugf("   INSERT into %s: %#v\n", d.table, s)
 		}
 	}
-	fmt.Fprint(d.out, buf.String())
```

Comment:
Printing out here prevents interleaving with other threads. Are you sure you want to allow interleaving?

Reply:
I'll go back to the buffer but log the output with logrus, main goal was just to have control over whether or not I see these verbosity.